### ISSUE TITLE
Enhance Curve Editor

### ISSUE THREAD

**Issue by TheDuriel** [godotengine/godot-proposals#48]
**Describe the project you are working on:**
I'm working on a Utility driven AI system for a Sims like game. It utilized mathematical functions (curves) to evaluate which actions to take. Godots curve resource is very useful here, as it removes the need to write the function directly through code, and provides a visual representation of the range.

**Describe how this feature / enhancement will help your project:**

Unlike the gradient editor for example, the Curve editor only provides points and handles. Making it impossible to set accurate values. The gradient editor exposes the array for each entry, allowing you to set the precise position of each point. I would like to have the same ability for points and their handles in the curve editor. (Making it much easier for example to create a logistic or logit curve)

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**

**Describe implementation detail for your proposal (in code), if possible:**

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Technically all these functions could be written in code. However using curves you can reduce the code to just two lines and get the benefit of a visual representation.
![image](https://user-images.githubusercontent.com/44248915/64487830-cc11c500-d23f-11e9-9363-fa812a6cee2f.png)

**Is there a reason why this should be core and not an add-on in the asset library?:**
It modifies the Inspector.

**Comment by Xrayez** [comment]
I'm using curve editors for fine-tuning water parallax layers to create correct illusion of far and near layers that way for depth, and in some cases it needs to be precise to achieve this illusion.

![water_params_curve](https://user-images.githubusercontent.com/17108460/64488765-506e4300-d254-11e9-802a-531cf69186b9.png)

**Comment by WiggleWizard** [comment]
Improved accuracy (mentioned in the main ticket) and scaling would be nice to have: the ability to set min and max on both X and Y would help a lot too, to avoid having to post scale interpolated values.

**Comment by WiggleWizard** [comment]
So I've been working on an advanced curve editor and have a quick question.

In the GIF below where the red dot represents interpolation along the tangent while the green one represents the line - curve intersection; which interpolation is preferred/correct? 

![curve](https://user-images.githubusercontent.com/1405402/89697567-91fbe800-d914-11ea-8031-f20728b0ffe8.gif)

I'm of the mind that both should be offered as an option but default to tangent interpolation for legacy support? Line - curve intersection (green dot) makes more sense to me as the curve the developer is creating a sort of graph where an X lookup results in the corresponding Y.

Thoughts?

**Comment by Inferno630** [comment]
Specifying exact values for points and their tangents should also be added to Curve2D and Curve3D. All the Curve resources are a bit hamstrung without this ability.

**Comment by fire-forge** [comment]
This would make working with curves much easier. It could be implemented similarly to Gradient, where the data is hidden away in a "Raw Data" section unless you need it.
